app/src/main/java/com/cmput301f20t21/bookfriends/ui/browse/BrowseFragment.java

```diff
@@ -80,8 +76,19 @@ public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
         inflater.inflate(R.menu.browse_search_menu, menu);
         super.onCreateOptionsMenu(menu, inflater);
         searchView = (SearchView) menu.findItem(R.id.book_search_bar).getActionView();
-    }
-    private void inflateSearchedList() {
-        fragmentManager = getChildFragmentManager();
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                vm.filterBookWithKeyword(query);
+                return false;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String newText) {
+                vm.filterBookWithKeyword(newText);
+                return false;
```

Comment:
based on the [OnQueryTextListener documentation](https://developer.android.com/reference/android/widget/SearchView.OnQueryTextListener#onQueryTextChange(java.lang.String)), returning false means letting the searchView perform its default action. Since we have custom search function, I think we should return true here both both functions?